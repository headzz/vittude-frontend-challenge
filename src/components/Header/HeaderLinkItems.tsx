import Link from 'next/link'
import { ListHeaderOptions } from './styled'
import { HomeIcon } from '@components/Icon/Home'
import { useTheme } from 'styled-components'
import useResizeScreen from '@hooks/useResizeScreen'

export const HeaderLinkItems = () => {
  const theme = useTheme()
  const { isMobile } = useResizeScreen()
  const svgColor = isMobile ? theme.colors.dark[200] : theme.colors.light[100]
  return (
    <ListHeaderOptions>
      <li>
        <Link href="https://www.vittude.com/">Site Institucional</Link>
      </li>
      <li>
        <Link href="https://www.vittude.com/blog/">Blog</Link>
      </li>
      <li>
        <Link href="/">
          <HomeIcon fill={svgColor} />
        </Link>
      </li>
    </ListHeaderOptions>
  )
}
