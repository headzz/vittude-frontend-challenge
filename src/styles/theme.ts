export const theme = {
  colors: {
    primary: '#FF6952',
    secondary: '#F26B5B',
    error: '#F63D15',
    dark: {
      100: '#777C88',
      200: '#2E3035',
      300: '#000000',
    },
    light: {
      100: '#FFFFFF',
      200: '#E7EAF4',
    },
  },
  font: {
    size: {
      normal: '1rem',
      small: '0.875rem',
      large: '2rem',
      extraLarge: '3rem',
    },
    type: { header: 'Inter', normal: 'Fira Sans' },
  },
  spacing: {
    extraSmall: '10px',
    small: '12px',
    normal: '16px',
    large: '24px',
    extraLarge: '48px',
  },
}
