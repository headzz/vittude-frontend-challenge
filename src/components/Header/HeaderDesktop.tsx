import Image from 'next/image'
import { HeaderLinkItems } from './HeaderLinkItems'
import { HeaderStyled, LogoLink } from './styled'

export const HeaderDesktop = () => {
  return (
    <HeaderStyled>
      <LogoLink
        href="https://www.vittude.com/"
        aria-label="redirecionamento para pÃ¡gina institucional"
      >
        <Image
          src="static/vittude-logo.svg"
          alt="logo da empresa vittude"
          width={107}
          height={32}
        />
      </LogoLink>
      <HeaderLinkItems />
    </HeaderStyled>
  )
}
