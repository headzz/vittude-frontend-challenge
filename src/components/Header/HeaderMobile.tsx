import { HamburguerMenu } from '@components/HamburgerMenu'
import { HeaderStyled } from './styled'
import { HeaderLinkItems } from './HeaderLinkItems'
import Image from 'next/image'

export const MobileHeader = () => {
  return (
    <HeaderStyled>
      <HamburguerMenu>
        <HeaderLinkItems />
      </HamburguerMenu>
      <Image
        src="/static/vittude-logo.svg"
        alt="vittude logo"
        width={107}
        height={32}
      />
    </HeaderStyled>
  )
}
