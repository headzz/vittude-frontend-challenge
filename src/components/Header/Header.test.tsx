import { fireEvent, render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from '@styles/theme'
import { HeaderDesktop } from './HeaderDesktop'
import { HeaderLinkItems } from './HeaderLinkItems'
import { MobileHeader } from './HeaderMobile'

describe('Header', () => {
  it('renders header link items', () => {
    render(
      <ThemeProvider theme={theme}>
        <HeaderLinkItems />,
      </ThemeProvider>,
    )

    const list = screen.getByRole('list')

    const site = screen.getByRole('link', {
      name: /site institucional/i,
    })

    const blog = screen.getByRole('link', {
      name: /blog/i,
    })

    const home = screen.getByRole('graphics-symbol')

    expect(site).toBeInTheDocument()
    expect(blog).toBeInTheDocument()
    expect(home).toBeInTheDocument()
    expect(list).toMatchSnapshot()
  })

  it('renders header desktop layout', () => {
    render(
      <ThemeProvider theme={theme}>
        <HeaderDesktop />,
      </ThemeProvider>,
    )

    const header = screen.getByRole('banner')

    const logo = screen.getByAltText('logo da empresa vittude')

    const site = screen.getByRole('link', {
      name: /site institucional/i,
    })

    const blog = screen.getByRole('link', {
      name: /blog/i,
    })

    const home = screen.getByRole('graphics-symbol')

    expect(logo).toBeInTheDocument()
    expect(site).toBeInTheDocument()
    expect(blog).toBeInTheDocument()
    expect(home).toBeInTheDocument()
    expect(header).toMatchSnapshot()
  })

  it('renders header mobile layout', () => {
    render(
      <ThemeProvider theme={theme}>
        <MobileHeader />,
      </ThemeProvider>,
    )
    const button = screen.getByRole('button')
    fireEvent.click(button)
    const nav = screen.getByRole('navigation')

    const header = screen.getByRole('banner')

    const logo = screen.getByRole('img')

    const site = screen.getByRole('link', {
      name: /site institucional/i,
    })

    const blog = screen.getByRole('link', {
      name: /blog/i,
    })

    const home = screen.getByRole('graphics-symbol')

    expect(logo).toBeInTheDocument()
    expect(site).toBeInTheDocument()
    expect(blog).toBeInTheDocument()
    expect(home).toBeInTheDocument()
    expect(header).toMatchSnapshot()
    expect(nav).toMatchSnapshot()
  })
})
