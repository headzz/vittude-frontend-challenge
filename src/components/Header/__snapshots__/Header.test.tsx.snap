// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header desktop layout 1`] = `
<header
  class="sc-bcPKhP hjiQzg"
>
  <a
    aria-label="redirecionamento para pÃ¡gina institucional"
    class="sc-dkjKgF ftvYuD"
    href="https://www.vittude.com/"
  >
    <img
      alt="logo da empresa vittude"
      data-nimg="1"
      decoding="async"
      height="32"
      loading="lazy"
      src="static/vittude-logo.svg"
      style="color: transparent;"
      width="107"
    />
  </a>
  <ul
    class="sc-grXZZQ iaSPMK"
  >
    <li>
      <a
        href="https://www.vittude.com/"
      >
        Site Institucional
      </a>
    </li>
    <li>
      <a
        href="https://www.vittude.com/blog/"
      >
        Blog
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        <svg
          fill="none"
          height="24"
          role="graphics-symbol"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_1_208)"
            id="Social/public"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM4 12C4 11.39 4.08 10.79 4.21 10.22L8.99 15V16C8.99 17.1 9.89 18 10.99 18V19.93C7.06 19.43 4 16.07 4 12ZM17.89 17.4C17.63 16.59 16.89 16 15.99 16H14.99V13C14.99 12.45 14.54 12 13.99 12H7.99V10H9.99C10.54 10 10.99 9.55 10.99 9V7H12.99C14.09 7 14.99 6.1 14.99 5V4.59C17.92 5.77 20 8.65 20 12C20 14.08 19.19 15.98 17.89 17.4Z"
              fill="#FFFFFF"
              id="Vector"
            />
          </g>
          <defs>
            <clippath
              id="clip0_1_208"
            >
              <rect
                fill="#FFFFFF"
                height="24"
                width="24"
              />
            </clippath>
          </defs>
        </svg>
      </a>
    </li>
  </ul>
</header>
`;

exports[`Header renders header link items 1`] = `
<ul
  class="sc-grXZZQ iaSPMK"
>
  <li>
    <a
      href="https://www.vittude.com/"
    >
      Site Institucional
    </a>
  </li>
  <li>
    <a
      href="https://www.vittude.com/blog/"
    >
      Blog
    </a>
  </li>
  <li>
    <a
      href="/"
    >
      <svg
        fill="none"
        height="24"
        role="graphics-symbol"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_1_208)"
          id="Social/public"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM4 12C4 11.39 4.08 10.79 4.21 10.22L8.99 15V16C8.99 17.1 9.89 18 10.99 18V19.93C7.06 19.43 4 16.07 4 12ZM17.89 17.4C17.63 16.59 16.89 16 15.99 16H14.99V13C14.99 12.45 14.54 12 13.99 12H7.99V10H9.99C10.54 10 10.99 9.55 10.99 9V7H12.99C14.09 7 14.99 6.1 14.99 5V4.59C17.92 5.77 20 8.65 20 12C20 14.08 19.19 15.98 17.89 17.4Z"
            fill="#FFFFFF"
            id="Vector"
          />
        </g>
        <defs>
          <clippath
            id="clip0_1_208"
          >
            <rect
              fill="#FFFFFF"
              height="24"
              width="24"
            />
          </clippath>
        </defs>
      </svg>
    </a>
  </li>
</ul>
`;

exports[`Header renders header mobile layout 1`] = `
<header
  class="sc-bcPKhP hjiQzg"
>
  <button
    class="sc-hLclGa ftZzDx"
    open=""
  >
    <span
      class="sc-eDnVMP eMeRph"
      open=""
    />
    <span
      class="sc-eDnVMP eMeRph"
      open=""
    />
  </button>
  <nav
    class="sc-jSwlEQ guTZrn"
    open=""
  >
    <ul
      class="sc-grXZZQ iaSPMK"
    >
      <li>
        <a
          href="https://www.vittude.com/"
        >
          Site Institucional
        </a>
      </li>
      <li>
        <a
          href="https://www.vittude.com/blog/"
        >
          Blog
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          <svg
            fill="none"
            height="24"
            role="graphics-symbol"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_1_208)"
              id="Social/public"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM4 12C4 11.39 4.08 10.79 4.21 10.22L8.99 15V16C8.99 17.1 9.89 18 10.99 18V19.93C7.06 19.43 4 16.07 4 12ZM17.89 17.4C17.63 16.59 16.89 16 15.99 16H14.99V13C14.99 12.45 14.54 12 13.99 12H7.99V10H9.99C10.54 10 10.99 9.55 10.99 9V7H12.99C14.09 7 14.99 6.1 14.99 5V4.59C17.92 5.77 20 8.65 20 12C20 14.08 19.19 15.98 17.89 17.4Z"
                fill="#FFFFFF"
                id="Vector"
              />
            </g>
            <defs>
              <clippath
                id="clip0_1_208"
              >
                <rect
                  fill="#FFFFFF"
                  height="24"
                  width="24"
                />
              </clippath>
            </defs>
          </svg>
        </a>
      </li>
    </ul>
  </nav>
  <img
    alt="vittude logo"
    data-nimg="1"
    decoding="async"
    height="32"
    loading="lazy"
    src="/static/vittude-logo.svg"
    style="color: transparent;"
    width="107"
  />
</header>
`;

exports[`Header renders header mobile layout 2`] = `
<nav
  class="sc-jSwlEQ guTZrn"
  open=""
>
  <ul
    class="sc-grXZZQ iaSPMK"
  >
    <li>
      <a
        href="https://www.vittude.com/"
      >
        Site Institucional
      </a>
    </li>
    <li>
      <a
        href="https://www.vittude.com/blog/"
      >
        Blog
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        <svg
          fill="none"
          height="24"
          role="graphics-symbol"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_1_208)"
            id="Social/public"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM4 12C4 11.39 4.08 10.79 4.21 10.22L8.99 15V16C8.99 17.1 9.89 18 10.99 18V19.93C7.06 19.43 4 16.07 4 12ZM17.89 17.4C17.63 16.59 16.89 16 15.99 16H14.99V13C14.99 12.45 14.54 12 13.99 12H7.99V10H9.99C10.54 10 10.99 9.55 10.99 9V7H12.99C14.09 7 14.99 6.1 14.99 5V4.59C17.92 5.77 20 8.65 20 12C20 14.08 19.19 15.98 17.89 17.4Z"
              fill="#FFFFFF"
              id="Vector"
            />
          </g>
          <defs>
            <clippath
              id="clip0_1_208"
            >
              <rect
                fill="#FFFFFF"
                height="24"
                width="24"
              />
            </clippath>
          </defs>
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;
